bpf/bpf_lb.c

```diff
@@ -102,18 +121,40 @@ static inline int handle_ipv6(struct __sk_buff *skb)
 		return DROP_NO_SERVICE;
 
 	ipv6_addr_copy(&new_dst, &svc->target);
+
+#ifndef ENCAP_IFINDEX
+	/* Do not store revnat index in address if encap mode is on */
 	if (svc->rev_nat_index)
 		new_dst.p4 |= svc->rev_nat_index;
+#endif
 
 	ret = lb6_xlate(skb, &new_dst, nexthdr, l3_off, l4_off, &csum_off, &key, svc);
 	if (IS_ERR(ret))
 		return ret;
 
-	return TC_ACT_REDIRECT;
+#ifdef ENCAP_IFINDEX
+	ret = lb6_encap(skb, &new_dst, svc->rev_nat_index);
+	if (IS_ERR(ret))
```

Comment:
I had `goto drop` before I rebased with @mchalla's changes and that change got lost.